src/Engine/ProtoCore/Utils/ArrayUtils.cs

```diff
@@ -1051,6 +1051,28 @@ public static StackValue[] GetKeys(StackValue array, RuntimeCore runtimeCore)
         }
 
         /// <summary>
+        /// Get a list of key-value pairs for an array.
+        /// </summary>
+        /// <param name="array"></param>
+        /// <param name="runtimeCore"></param>
+        /// <returns></returns>
+        public static IEnumerable<KeyValuePair<StackValue, StackValue>> GetKeyValuePairs(StackValue array, RuntimeCore runtimeCore)
```

Comment:
Should we return IDictionary<StackValue, StackValue> so this method can be used in other context as well.